src/policy/protect_payload.rs

```diff
@@ -212,3 +215,22 @@ impl ProtectPayloadPolicy {
         }
     }
 }
+
+// ———————————————————————————————— Hash primitive ———————————————————————————————— //
+
+#[allow(dead_code)]
+fn hash_memory_range(start: usize, end: usize) -> [u8; 32] {
+    assert!(start <= end, "Invalid memory range for payload hashing");
+
+    unsafe {
+        let data = slice::from_raw_parts(start as *const u8, end - start);
+
+        let mut hasher = Sha3::v256();
+        hasher.update(data);
```

Comment:
I don't think we need to hash the registers, because between the firmware and the palyoad we have the policy that can filter them. The hash is here because we need to make sure the payload is correctly loaded. Do you see what I mean?